proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1203 @@ If WiFi credentials are automatically generated by the IVI system, the password
 
 ## Proposed solution
 
-The available transport type for Secondary Transport can be configured through `smartDeviceLink.ini` file. 
+The functionality to be introduced in the current proposal mostly relies on the IVI and mobile device capabilities. It is proposed to introduce a new capability type to be specific for networking purposes to both Mobile and HMI API:
 
-SDL adapters could define supported WiFi modes in the `smartDeviceLink.ini` file. The mobile proxy could use the defined RPCs below to share WiFi credentials and connect automatically with users' consent.  
+### HMI_API
 
-If the IVI supports WiFi access point mode, the Android and iOS libraries provide APIs to connect to a WiFi network in case WiFi SSID and password is known. These APIs could be used in a mobile proxy to connect to the WiFi access point created by the HMI. The flow for the proposed solution would be:
+SDL should be able to receive networking capabilities with `UI.GetCapabilities` request at the very beginning of SDL - HMI communication. If omitted, HMI will be considered as not suitable for the feature initialization.
 
-`HMI creates WiFi Access Point --> HMI provides WiFi Access point credentials to SDL Core --> SDL Core shares WiFi credentials using Primary Transport --> Mobile Proxy receives WiFi access point credentials  --> Mobile Proxy automatically connects to the WiFi access point.`
+```xml
+<enum name="NetworkHost">
+    <element name="MOBILE"/>
+    <element name="VEHICLE"/>
+    <element name="EXTERNAL"/>

```

Comment:
Can't remove this from the solution because it will create several issues that was not discussed on the workshop. 
There are logic depend on this value.

Reply:
At this time, the proposal needs to be revised based on the decision made by the Steering Committee, which does not include this element.  Please remove.  If Luxoft has found that there's a need for this element, you can comment on the review issue with this information/requested change during the Steering Committee's next review.

Reply:
Fixed proposal to keep bounds agreed on workshop and https://github.com/smartdevicelink/sdl_evolution/issues/799#issuecomment-588434670